std/src/std/iter.inko

```diff
@@ -301,6 +301,19 @@ trait pub Iter[T] {
     }
   }
 
+  # Join two `Iter` objects together, one after another.
+  #
+  # # Examples
+  #
+  # ```inko
+  # let a = [10, 20, 30]
+  # let b = [40, 50, 60]
+  # a.iter.chain(b.iter).to_array == [10, 20, 30, 40, 50, 60] # => true
+  # ```
+  fn pub move chain[I: mut + Iter[T]](other: I) -> Stream[T] {
+    Stream.new(fn move { self.next.else(fn { other.next }) })
```

Comment:
This will always first advance the iterator on the left, then the one on the right. If the iterator tries to do e.g. IO behind the scenes, this can be pretty costly. 

A better approach is to keep track of a variable `iter_left` that is initially `true`, and when `true` the closure calls `self.next`. If this returns `None`, the flag is set to `false`, the result of `other.next` is returned, and future calls of the closure instead call `other.next` directly.

Reply:
OK, that's done.